### DISCUSSION TITLE
Pull requests merged through the API are not auto closing issues (closing keywords)

### DISCUSSION THREAD

**Discussion by phstc** [community/community#24869] [Apps, API and Webhooks]
<p>Hi</p>
<p><a href="https://github.com/marketplace/merge-when-green" rel="nofollow noopener">I have a probot app</a> that merges pull requests once all CI statuses succeed. </p>
<p><a href="https://github.com/phstc/probot-merge-when-green/issues/5" rel="nofollow noopener">Some users reported</a> that the pull requests merged with the bot are not triggering <a href="https://help.github.com/en/articles/closing-issues-using-keywords" rel="nofollow noopener">closing issues with keywords</a>.</p>
<p>I could reproduce it with <a href="https://github.com/phstc/probot-merge-when-green/pull/9" rel="nofollow noopener">this pull request</a> and <a href="https://github.com/phstc/probot-merge-when-green/issues/8" rel="nofollow noopener">this issue</a>.</p>
<p>Is closing via API supposed to behave as closing via web? </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/f/f99769b2ab16b9670a06d79132dab9af3ccea6de.png" data-download-href="/uploads/short-url/zBZi1D1DgkjlMXvTZKUSDkAmmjs.png?dl=1" title="Screen Shot 2019-06-15 at 7.30.33 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/105652/181087121-cf850d10-c66a-4868-b736-013524d85a9b.png" alt="Screen Shot 2019-06-15 at 7.30.33 PM.png" data-base62-sha1="zBZi1D1DgkjlMXvTZKUSDkAmmjs" width="622" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/f/f99769b2ab16b9670a06d79132dab9af3ccea6de_2_622x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/f/f99769b2ab16b9670a06d79132dab9af3ccea6de_2_933x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/f/f99769b2ab16b9670a06d79132dab9af3ccea6de_2_1244x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/f/f99769b2ab16b9670a06d79132dab9af3ccea6de_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-06-15 at 7.30.33 PM.png</span><span class="informations">1578×1268 254 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/b3f3ba12cf5a497e0a715713f31763a4dd1691de.png" data-download-href="/uploads/short-url/pFVEgYqQIJ8a2mAfX5A0x2oljrM.png?dl=1" title="Screen Shot 2019-06-15 at 7.30.49 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/105652/181087123-bf37a0d2-5f32-4464-8ffc-36f5ec6e5287.png" alt="Screen Shot 2019-06-15 at 7.30.49 PM.png" data-base62-sha1="pFVEgYqQIJ8a2mAfX5A0x2oljrM" width="690" height="394" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/b3f3ba12cf5a497e0a715713f31763a4dd1691de_2_690x394.png, https://aws1.discourse-cdn.com/github/optimized/2X/b/b3f3ba12cf5a497e0a715713f31763a4dd1691de_2_1035x591.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/b/b3f3ba12cf5a497e0a715713f31763a4dd1691de_2_1380x788.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/b3f3ba12cf5a497e0a715713f31763a4dd1691de_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-06-15 at 7.30.49 PM.png</span><span class="informations">1640×938 98 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by phstc** [comment, marked as answer]
<p>The mystery has been solved :tada:</p>
<p>After playing back-and-forth, even trying some non-sense stuff. I found the problem!</p>
<p>\ <img src="https://user-images.githubusercontent.com/105652/181087127-7db29e9b-92cf-4287-901a-11e26838e418.gif" alt="giphy" data-base62-sha1="2psjKOQp821ABCbfE0fZuUp6PRM" width="350" height="149"></p>
<p>The bot needs issue write permissions, otherwise <a href="https://help.github.com/en/articles/closing-issues-using-keywords" rel="nofollow noopener">closing issues using keywords</a> will “fail silently” (╯°□°)╯︵ ┻━┻</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hey <a class="mention" href="/phstc">@phstc</a>,</p>
<p>Thanks a lot for being here, we appreciate you coming back to share your solution.</p>

**Comment by mboes** [comment]
<p>We’re observing the same issue, but with the Mergify app. Which does seem to have issue write permission, so I’m stumped as to how to solve this…</p>

**Comment by phstc** [comment]
<p>Hi <a class="mention" href="/mboes">@mboes</a> </p>
<p>If the problem is with permissions you may be able to see some specific errors in the logs.</p>

**Comment by mboes** [comment]
<p>Turns out this was the same problem as for you. It’s been fixed by upstream since a couple of months ago. See <a href="https://github.com/Mergifyio/mergify-engine/issues/466#issuecomment-520297764" rel="nofollow noopener">https://github.com/Mergifyio/mergify-engine/issues/466#issuecomment-520297764</a>.</p>
